iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -326,6 +327,11 @@ cxx::expected<int32_t, IpcChannelError> UnixDomainSocket::createSocket(const Ipc
     // initialize the sockAddr data structure with the provided name
     memset(&m_sockAddr, 0, sizeof(m_sockAddr));
     m_sockAddr.sun_family = AF_LOCAL;
+    const uint64_t maxDestinationLength = cxx::strlen2(m_sockAddr.sun_path);
+    if (m_name.length() > maxDestinationLength)
+    {
+        return cxx::error<IpcChannelError>(IpcChannelError::INVALID_CHANNEL_NAME);
```

Comment:
How do we want to test this? Currently the UnixDomainSocket uses the same tests as the MessageQueue and is only tested on maxOS.
Shall I make the MessageQueue test a typed test and add a test for UnixDomainSocket specific behavior?

Reply:
I would say this requires a set of separate tests specific to the UnixDomainSocket. To make a typed test reasonable the types would always need to have the same interface and semantics (the message queue will not have a createSocket method). E.g. are the maximum allowed name length the same? Otherwise both would behave differently, unless some additional measures are implemented (minimum of both allowed length etc.).

So just one UnixDomainSocket test were the name is (just) short enough and another one were it is too long and construction fails would be my proposal.

Those tests could (and should, if we provide the UnixDomainSocket class in utils) also run on Linux I guess, they are in the POSIX standard if I am not mistaken.

Reply:
Creating additional tests unrelated to string is not part of this pull request in my opinion.

Reply:
Well, currently the MessageQueue tests are used for the UnixDomainSocket on macOS, so there is a common interface and semantics. There are just fully equal, like with the name length.

Reply:
>
> Shall I make the MessageQueue test a typed test 

Would that take lots of effort?
In the end we want to have universal tests that run on every system and not just on a specific one, don't we?

Reply:
No, I guess that wouldn't be much afford

Reply:
We already differentiate in the message queue test https://github.com/eclipse/iceoryx/blob/master/iceoryx_utils/test/moduletests/test_message_queue.cpp#L163

So you can add a test that is specific for UDS?